.github/compare-assets/index.js

```diff
@@ -82,24 +82,55 @@ const runner = async () => {
 			}
 		);
 
-		await octokit.rest.issues.createComment( {
+		const currentComments = await octokit.rest.issues.listComments( {
 			owner,
 			repo,
 			issue_number: payload.pull_request.number,
-			body:
-				'# Script Dependencies Report' +
-				'\n\n' +
-				'The `compare-assets` action has detected some changed script dependencies between this branch and ' +
-				'trunk. Please review and confirm the following are correct before merging.' +
-				'\n\n' +
-				'| Script Handle | Added | Removed |    |' +
-				'\n' +
-				'| ------------- | ------| ------- | -- |' +
-				'\n' +
-				reportContent +
-				'\n\n' +
-				'__This comment was automatically generated by the `./github/compare-assets` action.__',
 		} );
+
+		let reportCommentId;
+		if (
+			Array.isArray( currentComments.data ) &&
+			currentComments.data.length > 0
+		) {
+			const comment = currentComments.data.find( ( comment ) =>
+				comment.body.includes( 'Script Dependencies Report' )
```

Comment:
Minor concern is that if someone says this phrase that comment will be used, maybe we can add something hidden in the report and use that? That's not a blocker and I think this PR is great, Thank you!!

Reply:
> maybe we can add something hidden in the report and use that?

GitHub doesn't support storing metadata to its objects. Even if we can hack it to store metadata in the comment body, we're still parsing the comment body. It also makes the condition more complex, which isn't worth the effort IMO.